filebeat/harvester/reader/docker_json.go

```diff
@@ -41,16 +41,19 @@ func NewDockerJSON(r Reader, stream string) *DockerJSON {
 // parseCRILog parses logs in CRI log format.
 // CRI log format example :
 // 2017-09-12T22:32:21.212861448Z stdout 2017-09-12 22:32:21.212 [INFO][88] table.go 710: Invalidating dataplane cache
-func parseCRILog(b []byte, msg *crioLog, message Message) (Message, error) {
-	log := string(b)
-	ts, err := time.Parse(time.RFC3339, strings.Fields(log)[0])
+func parseCRILog(message Message, msg *crioLog) (Message, error) {
+	log := strings.SplitN(string(message.Content), " ", 3)
+	if len(log) < 3 {
+		return message, errors.New("invalid CRI log")
+	}
+	ts, err := time.Parse(time.RFC3339, log[0])
 	if err != nil {
 		return message, errors.Wrap(err, "parsing CRI timestamp")
 	}
 
-	stream := strings.Fields(log)[1]
+	stream := log[1]
 	// Anything after stream.
-	logMessage := strings.Fields(log)[2:]
+	logMessage := log[2:]
 	msg.Timestamp = ts
 	msg.Stream = stream
 	msg.Log = []byte(strings.Join(logMessage, " "))
```

Comment:
You no longer need `strings.Join` here, this should be just log[2]

Reply:
True, no need for `strings.Join` , but you meant `msg.Log = []byte(logMessage[0])` right ?

Reply:
yep, or remove `logMessage` variable and just do `msg.Log = []byte(log[2]))`